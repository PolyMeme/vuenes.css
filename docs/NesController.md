# nes-controller 

## props 

- `type` ***String*** (*required*) 

- `jp` ***Boolean*** (*optional*) 

## computed properties 

- `controllerClasses` 

   **dependencies:** `type`, `jp` 


