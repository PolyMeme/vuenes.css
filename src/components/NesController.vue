<template>
  <li :class="controllerClasses" />
</template>

<script>
/**
 * Vue Component implementation of the Nes Controller
 * @author Jesus Fernando Alvarez Franco
 * @license MIT
 */
export default {
  props: {
    type: {
      /**
       * Define the type of controller. It must be `nes` or `snes`
       */
      type: String,
      required: true,
      validate (input) {
        return ['nes', 'snes'].includes(input)
      }
    },
    /**
     * Define if the controller is the japanese one
     */
    jp: Boolean
  },
  computed: {
    /**
     * Calculates the class to be used for the controlled depending of the props
     * @returns {{[p: string]: boolean}}
     */
    controllerClasses () {
      const cssClass = `${this.type}-${this.jp ? 'jp-' : ''}logo`

      return {
        [cssClass]: true
      }
    }
  }
}
</script>
