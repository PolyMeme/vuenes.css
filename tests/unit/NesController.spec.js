import { mount } from '@vue/test-utils'
import NesController from '../../src/components/NesController.vue'

describe('NesController.vue', () => {
  const createControllerWrapper = (propsData = {}) => {
    return mount(NesController, {
      propsData
    })
  }

  it('should render the nes controller properly', () => {
    const wrapper = createControllerWrapper({
      type: 'nes'
    })

    expect(wrapper.exists()).toBe(true)
    expect(wrapper.props('type')).toBe('nes')
    expect(wrapper.find('li.nes-logo').exists()).toBe(true)
  })

  it('should render the snes controller properly', () => {
    const wrapper = createControllerWrapper({
      type: 'snes'
    })

    expect(wrapper.exists()).toBe(true)
    expect(wrapper.props('type')).toBe('snes')
    expect(wrapper.find('li.snes-logo').exists()).toBe(true)
  })

  it('should render the japanese nes controller properly', () => {
    const wrapper = createControllerWrapper({
      type: 'nes',
      jp: true
    })

    expect(wrapper.exists()).toBe(true)
    expect(wrapper.props('type')).toBe('nes')
    expect(wrapper.find('li.nes-jp-logo').exists()).toBe(true)
  })

  it('should render the japanese snes controller properly', () => {
    const wrapper = createControllerWrapper({
      type: 'snes',
      jp: true
    })

    expect(wrapper.exists()).toBe(true)
    expect(wrapper.props('type')).toBe('snes')
    expect(wrapper.find('li.snes-jp-logo').exists()).toBe(true)
  })

  it('should fail prop validation', () => {
    const wrapper = createControllerWrapper({
      type: 'invalid'
    })

    expect(wrapper.vm.$options.props.type.validate()).toBeFalsy()
  })

  it('should match snapshot', () => {
    const wrapper = createControllerWrapper({
      type: 'snes',
      jp: true
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
