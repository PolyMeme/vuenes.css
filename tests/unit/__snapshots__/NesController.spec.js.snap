// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NesController.vue should match snapshot 1`] = `<li class="snes-jp-logo"></li>`;
